pkg/controller/installation/core_controller.go

```diff
@@ -1054,17 +1052,21 @@ func (r *ReconcileInstallation) Reconcile(ctx context.Context, request reconcile
 		}
 	}
 
-	certificateManager, err := certificatemanager.Create(r.client, &instance.Spec, r.clusterDomain, common.OperatorNamespace())
+	certificateManager, err := certificatemanager.CreateWithOptions(r.client, &instance.Spec, r.clusterDomain, common.OperatorNamespace(), certificatemanager.WithLogger(reqLogger))
 	if err != nil {
 		r.status.SetDegraded(operator.ResourceCreateError, "Unable to create the Tigera CA", err, reqLogger)
 		return reconcile.Result{}, err
 	}
+
+	// The manager internal TLS certificate is needed by kube-controllers in order to establish mTLS with Voltron
+	// in management clusters. kube-controllers copies the es-gateway and linseed certificates to managed clusters.
+	// TODO: We should ideally move the copying of these secrets elsehwere, since kube-controllers isn't installed in multi-tenant
+	// clusters right now. Without something doing this, fluentd in managed clusters won't be able to verify Voltron.
 	var managerInternalTLSSecret certificatemanagement.KeyPairInterface
 	if instance.Spec.Variant == operator.TigeraSecureEnterprise && managementCluster != nil {
-		dnsNames := append(dns.GetServiceDNSNames(render.ManagerServiceName, render.ManagerNamespace, r.clusterDomain), render.ManagerServiceIP)
-		managerInternalTLSSecret, err = certificateManager.GetOrCreateKeyPair(r.client, render.ManagerInternalTLSSecretName, common.OperatorNamespace(), dnsNames)
-		if err != nil {
-			r.status.SetDegraded(operator.CertificateError, fmt.Sprintf("Error ensuring internal manager TLS certificate %q exists and has valid DNS names", render.ManagerInternalTLSSecretName), err, reqLogger)
+		managerInternalTLSSecret, err = certificateManager.GetKeyPair(r.client, render.ManagerInternalTLSSecretName, common.OperatorNamespace())
```

Comment:
This keypair is now created by the manager controller - seemed weird that it was being created here. The logic for installing manager was thus leaking into other unrelated controllers.

Reply:
I don't think this kube controllers actually needs the manager tls cert. The regular kube controllers created by `kubecontrollers.NewCalicoKubeControllers` doesn't have the managedcluster controller enabled. That's in the es-kube-controllers.

This might be legacy code that was never cleaned up when we split out kube controllers, since none of the enabled controllers aren't for MCM. Might make your life a little easier if this is true...

Reply:
Yeah I think I agree with you. That's nice!